communitymanager/src/main/java/org/pesmypetcare/communitymanager/managers/ForumManagerClient.java

```diff
@@ -0,0 +1,216 @@
+        httpClient = new HttpClient();
+        gson = new Gson();
+    }
+
+    /**
+     * Creates a new forum.
+     *
+     * @param parentGroup The parent group name
+     * @param forum The forum data
+     * @throws MyPetCareException When the request fails
+     */
+    public void createForum(String parentGroup, ForumData forum) throws MyPetCareException {
+        httpClient.post(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup), null, null, gson.toJson(forum));
+    }
+
+    /**
+     * Deletes a forum.
+     *
+     * @param parentGroup The parent group name
+     * @param forumName The forum name
+     * @throws MyPetCareException When the request fails
+     */
+    public void deleteForum(String parentGroup, String forumName) throws MyPetCareException {
+        HttpParameter[] params = new HttpParameter[1];
+        params[0] = new HttpParameter(FORUM_PARAMETER, forumName);
+        httpClient.delete(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup), params, null, null);
+    }
+
+    /**
+     * Retrieves a forum information.
+     *
+     * @param parentGroup The parent group name
+     * @param forumName The forum name
+     * @return The forum data
+     * @throws MyPetCareException When the request fails
+     */
+    public ForumData getForum(String parentGroup, String forumName) throws MyPetCareException {
+        HttpParameter[] params = new HttpParameter[1];
+        params[0] = new HttpParameter(FORUM_PARAMETER, forumName);
+        HttpResponse response = httpClient
+                .get(COMMUNITY_BASE_URL + HttpParameter.encode(parentGroup), params, null, null);
+        return gson.fromJson(response.asString(), ForumData.class);
+    }
+
+    /**
+     * Retrieves all forums information from a group.
+     *
+     * @param groupName The group name
+     * @return All data of the forums belonging to the group
+     * @throws MyPetCareException When the request fails
+     */
+    public List<ForumData> getAllForums(String groupName) throws MyPetCareException {
+        HttpResponse response = httpClient.get(COMMUNITY_BASE_URL + HttpParameter.encode(groupName), null, null, null);
+        Type listType = TypeToken.getParameterized(List.class, ForumData.class).getType();
+        return gson.fromJson(response.asString(), listType);
+    }
+
+    /**
+     * Updates the forum name.
+     *
+     * @param parentGroup The parent group name
+     * @param forumName The forum name
+     * @param newName The new name
+     * @throws MyPetCareException When the request fails
+     */
+    public void updateName(String parentGroup, String forumName, String newName) throws MyPetCareException {
+        HttpParameter[] params = new HttpParameter[1];
+        params[0] = new HttpParameter("newName", newName);
+        String group = HttpParameter.encode(parentGroup);
+        String forum = HttpParameter.encode(forumName);
+        httpClient.put(COMMUNITY_BASE_URL + group + "/" + forum, params, null, null);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "/" appears 6 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5538337)